calico/common.py

```diff
@@ -46,6 +46,21 @@
 # string used by the logger.
 SYSLOG_FORMAT_STRING = '{excname}[%(process)s]: %(module)s@%(lineno)d %(message)s'
 
+# White-list for the --protocol match criteria.  We allow the guaranteed
+# string shortcuts as well as int/string versions of the raw IDs.  We disallow
+# 0 because the kernel cannot match on it directly.
+KERNEL_PROTOCOLS = set(["tcp", "udp", "icmp", "icmpv6", "sctp", "udplite"])
+KERNEL_PROTOCOLS.update(xrange(1, 256))
+KERNEL_PROTOCOLS.update(intern(str(p)) for p in xrange(1, 256))
+
+# Protocols that support a port match in iptables.  We allow the name and
+# protocol number.
+KERNEL_PORT_PROTOCOLS = set(["tcp", 6,
+                             "udp", 17,
+                             "udplite", 136,
+                             "sctp", 132,
+                             "dccp", 33])
+
```

Comment:
I think you need "6" in the list as well as 6; you are allowing an integer protocol to be passed as a string above, but not here.